lms/static/js/ccx/async_grade_book_download.js

```diff
@@ -0,0 +1,137 @@
+var defineJs = window.define || RequireJS.define;
+
+defineJs("async_grade_book_download",
+  [
+    'backbone',
+    'underscore',
+    'gettext'
+  ],
+  function (Backbone, _, gettext) {
+    'use strict';
+    var GradeBookDownloadView = Backbone.View.extend({
+
+      events: {
+        "click a#ccx-calculate-grades-csv" : "asyncPrepareGradeBook"
+      },
+
+      initialize: function() {
+        // set interval to fetch list of files that are ready for download.
+        this.clearMessageBoxes();
+        this.showReadyGradeBooks();
+      },
+
+      render: function () {
+        return this;
+      },
+
+      reloadDownloadList: function() {
+        // refresh list of grade books, ready for download.
+        var self = this;
+        this.clearMessageBoxes();
+        // Fetch interval for grade book download request 40 sec.
+        var POLL_INTERVAL = 20000;
+        this.timer = setInterval(function() {
+          self.showReadyGradeBooks();
+        }, POLL_INTERVAL);
+      },
+
+      showReadyGradeBooks: function() {
+        // Get list of grade book files that are ready for download
+        var alreadyDownloadedUrl = $("#ccx_report-downloads-table").data('endpoint');
+        var self = this;
+        $.ajax({
+          dataType: 'json',
+          url: alreadyDownloadedUrl
```

Comment:
You are fetching data from the server and rendering it. Backbone is designed to handle this use case, using Backbone models, templates, and [Backbone.sync](http://backbonejs.org/#Sync). Please, please, use the framework to its fullest potential, rather than reimplementing it poorly.